File: challenges/4-fizz-buzz/go/joaomoraski/fizzBuzz.go
package main

import (
	"fmt"
	"log"
	"os"      // acessar os argumentos da linha de comando (igual em C, 0 é o nome)
	"strconv" // conversoes
)

func FizzBuzz(numero int) (resultado []string) {
	const (
		FIZZ     string = "Fizz"
		BUZZ     string = "Buzz"
		FIZZBUZZ string = "FizzBuzz"
	)
	for i := 1; i <= numero; i++ {
		if i%3 == 0 {
			if i%5 == 0 {
				resultado = append(resultado, FIZZBUZZ)
			} else {
				resultado = append(resultado, FIZZ)
			}
		} else if i%5 == 0 {
			if i%3 == 0 {
				resultado = append(resultado, FIZZBUZZ)
			} else {
				resultado = append(resultado, BUZZ)
			}
		} else {
			resultado = append(resultado, strconv.Itoa(i))
		}
	}
	return
}

func main() {
	// chamada por argumento
	if len(os.Args) != 2 {
		fmt.Println("Número de argumentos informado é maior ou menor do que o necessário")
		return
	}
	// Atoi igual no C
	numero, err := strconv.Atoi(os.Args[1])
	if err != nil {
		log.Fatalf("Não foi possivel converter '%v' para um inteiro", os.Args[1])
	}
	for _, resultado := range FizzBuzz(numero) {
		fmt.Println(resultado)
	}

}

File: challenges/4-fizz-buzz/javascript/RafaelPereira7L/fizzbuzz.js
const x = Number(process.argv[2]);

if(x <= 0) return console.log("Por favor, insira um número maior que 0.");

for (let i = 1; i <= x; i++) {
    if (i % 3 === 0 && i % 5 === 0) {
        console.log("FizzBuzz");
    } else if (i % 3 === 0) {
        console.log("Fizz");
    } else if (i % 5 === 0) {
        console.log("Buzz");
    } else {
        console.log(i);
    }
}